src/metadata/persistence/comments.clj

```diff
@@ -82,26 +79,39 @@
    Returns:
      A lazy sequence of detailed comment information."
   [commenter-id]
-  (-> (select* :comments)
-      (fields :id
-              [:owner_id :commenter]
-              :post_time
-              :retracted
-              :retracted_by
-              [:value :comment]
-              :deleted
-              :target_id
-              :target_type)
-      (where {:owner_id commenter-id})
-      select))
+  (let [select-cols [:id [:owner_id :commenter] :post_time :retracted :retracted_by [:value :comment] :deleted :target_id :target_type]
+        col-keys (mapv #(if (vector? %) (nth % 1) %) select-cols)
+        q (-> (comments-base-query select-cols)
+              (h/where [:= :owner_id commenter-id]))]
+    (plan/select! ds col-keys (sql/format q))))
+
+(defn insert-comment
+  "Inserts a comment into the comments table.
+
+   Parameters:
+     owner       - The authenticated user making the comment
+     target-id   - The UUID of the thing being commented on
+     target-type - The type of target (`analysis`|`app`|`data`|`user`)
+     comment     - The comment
+
+   Returns:
+     It returns a comment resource"
+  [owner target-id target-type comment]
+  (let [insert-vals (jsql/insert! ds
+                                  (t "comments")
+                                  {:owner_id    owner
+                                   :target_id   target-id
+                                   :target_type (jtypes/as-other target-type)
+                                   :value       comment})]
```

Comment:
Unlike the multi-insert, the single insert still takes a map of columns to values.